src/stability_sdk/api.py

```diff
@@ -108,82 +170,460 @@ def list_projects(context: 'Context') -> List['Project']:
+                generation.Prompt(
+                    artifact=generation.Artifact(
+                        type=generation.ARTIFACT_VIDEO,
+                        mime="video/mp4",
+                        binary=binary_data,
+                    )
+                )
+            ],
+            asset=generation.AssetParameters(
+                action=generation.ASSET_PUT,
+                project_id=proj.id,
+                use=generation.ASSET_USE_INPUT,
+            ),
+        )
+        results = self._context._run_request(self._context._asset, request)
+        if generation.ARTIFACT_TEXT in results:
+            return results[generation.ARTIFACT_TEXT][0]
+        raise Exception(f"Failed to store video asset for project {proj.id}")
 
-    def update(self, title:str=None, file_id:str=None, file_uri:str=None):
+    def update_project(context: 'Context', proj: 'Project', title: str = None, file_id: str = None, file_uri: str = None) -> None:
         file = project.ProjectAsset(
             id=file_id,
             uri=file_uri,
             use=project.PROJECT_ASSET_USE_PROJECT,
         ) if file_id and file_uri else None
-        
-        self._context._proj_stub.Update(project.UpdateProjectRequest(
-            id=self.id, 
+
+        context._proj_stub.Update(project.UpdateProjectRequest(
+            id=proj.id,
             title=title,
             file=file
         ))
 
+
+class LocalFileBackend(StorageBackend):
+    _projects_root = None
+
+    def __init__(self, context: 'Context', primary: bool = False, primary_fs: bool = True, projects_root = 'projects'):
+        super().__init__(context, primary, primary_fs = primary_fs)
+        LocalFileBackend._projects_root = projects_root
+
+    @staticmethod
+    def create_project(
+            context: 'Context',
+            title: str,
+            access: project.ProjectAccess = project.PROJECT_ACCESS_PRIVATE,
+            status: project.ProjectStatus = project.PROJECT_STATUS_ACTIVE,
+            proj_id_to_use: str = None
+    ) -> 'Project':
+        proj_id = proj_id_to_use if proj_id_to_use else str(uuid.uuid4())
+        proj_file_id = proj_id # Let's keep it the same as the proj_id for now
+        proj = {"id": proj_id,
+                "title": title,
+                "file": {"id": proj_file_id}}
+        output_path = os.path.join(LocalFileBackend._projects_root, proj_id, proj_file_id)
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+        with open(output_path, "w") as file:
+            json.dump(proj, file)
+        return Project(context, proj)
+
+    @staticmethod
+    def get_project(context: 'Context', id: str) -> 'Project':
+        input_path = os.path.join(LocalFileBackend._projects_root, id, id)
+        with open(input_path, "r") as file:
+            proj = json.load(file)
+        return Project(context, proj)
+
+    @staticmethod
+    def delete_project(context: 'Context', id: str):
+        if not id:
+            return
```

Comment:
Yes. I added this as a safeguard for the local filesystem implementation since I was about to make it delete a directory with `rmtree` and I wanted to increase my confidence that I can't end up deleting a root dir if the specified id is empty. Since I'm keeping the check for this reason anyway, I'll make it a ValueError.